nixos/modules/services/desktops/dunst.nix

```diff
@@ -0,0 +1,538 @@
+                };
+
+                historyLength = mkOption {
+                    type = types.int;
+                    default = 20;
+                    description = ''
+                        Maximum amount of notifications kept in history
+                    '';
+                };
+
+                showIndicators = mkOption {
+                    type = types.bool;
+                    default = true;
+                    description = ''
+                        Display indicators for URLs (U) and actions (A).
+                    '';
+                };
+
+                lineHeight = mkOption {
+                    type = types.int;
+                    default = 0;
+                    description = ''
+                        The height of a single line.  If the height is smaller
+                        than the font height, it will get raised to the font
+                        height.
+                        This adds empty space above and under the text.
+                    '';
+                };
+
+                separatorHeight = mkOption {
+                    type = types.int;
+                    default = 2;
+                    description = ''
+                        Draw a line of "separatpr_height" pixel height between
+                        two notifications.
+                        Set to 0 to disable.
+                    '';
+                };
+
+                padding = mkOption {
+                    type = types.int;
+                    default = 8;
+                    description = ''
+                        Padding between text and separator.
+                    '';
+                };
+
+                horizontalPadding = mkOption {
+                    type = types.int;
+                    default = 8;
+                    description = ''
+                        Horizontal padding.
+                    '';
+                };
+
+                separatorColor = mkOption {
+                    type = types.str;
+                    default = "frame";
+                    description = ''
+                        Define a color for the separator.
+                        possible values are:
+                         * auto: dunst tries to find a color fitting to the background;
+                         * foreground: use the same color as the foreground;
+                         * frame: use the same color as the frame;
+                         * anything else will be interpreted as a X color.
+                    '';
+                };
+
+                startupNotification = mkOption {
+                    type = types.bool;
+                    default = false;
+                    description = ''
+                        Print a notification on startup.
+                        This is mainly for error detection, since dbus
+                        (re-)starts dunst automatically after a crash.
+                    '';
+                };
+
+                dmenu = mkOption {
```

Comment:
I'm not even sure why there is such an option. I only "ported" the configuration file to a nix expression without thinking too much about the settings.